sdk/python/kfp/compiler/compiler.py

```diff
@@ -802,17 +802,22 @@ def _create_workflow(self,
       raise ValueError('Either specify pipeline params in the pipeline function, or in "params_list", but not both.')
 
     args_list = []
+    kwargs_dict = dict()
     signature = inspect.signature(pipeline_func)
-    for arg_name in signature.parameters:
+    for arg_name, arg in signature.parameters.items():
       arg_type = None
       for input in pipeline_meta.inputs or []:
         if arg_name == input.name:
           arg_type = input.type
           break
-      args_list.append(dsl.PipelineParam(sanitize_k8s_name(arg_name, True), param_type=arg_type))
+      param = dsl.PipelineParam(sanitize_k8s_name(arg_name, True), param_type=arg_type)
+      if arg.kind == inspect.Parameter.KEYWORD_ONLY:
```

Comment:
Maybe we can put all arguments in a dictionary and then just use [signature.bind](https://docs.python.org/3/library/inspect.html#inspect.Signature.bind) which will sort out what goes where?

Reply:
I don't think it would work with position-only arguments:

```python
def foo(a: str, /, b: str, *, c: str):
    print(a, b, c)

import inspect
sig = inspect.signature(foo)

kargs = {"a": "A", "b": "B", "c": "C"}
bound_arguments = sig.bind(**kwargs) # Error: "a" is positional-only
```

Reply:
Position-only arguments won't work with kwargs in signature.bind:

```python
def foo(a: str, /, b: str, *, c: str): ...
    
import inspect
sig = inspect.signature(foo)

kwargs = {"a": "A", "b": "B", "c": "C"}
bound_arguments = sig.bind(**kwargs) # Error: "a" is positional-only

args = ["A"]
kwargs = {"b": "B", "c": "C"}
bound_arguments = sig.bind(**kwargs) # ok, no error
```

So it seems to me that the easiest approach is to only bound keyword-only arguments to kwargs, and the rest arguments to args.